### ISSUE TITLE
Fish autocomplete on neovim (or vim) scp://

### ISSUE THREAD

**Issue by AndreaHasani** [fish-shell/fish-shell#10041]
Fish already has proper scp completions but they don't work on neovim scp://
I tried to implement this myself but can't figure out how to make the complete -c, work with a command argument defined. (such as neovim scp://(autocomplete).

This works with ssh (with autocomplete plugins) so I think it should be doable. Sorry for asking this here but I kinda tried to do it myself (and even get some help on chatgpt) but just can't figure out the fish syntax on the completions.

Thanks

**Comment by faho** [comment]
>even get some help on chatgpt

As a sidenote: Chatgpt (and other LLMs) are even worse than usual on a language that has as little a corpus of available code as fish script.

I heavily recommend against trying it on fish, and I have already had my time wasted by people opening PRs of LLM-generated code, and I would prefer not to have my time wasted.

----

Anyway, now that that's done:

This is referring to (neo)vim's "netrw" plugin/system/whatever.

It's theoretically *possible* to complete, but it would rely on us running ssh on the remove system to get paths, which *isn't* typically a neovim dependency.

Basically, what you would do is this:

```fish
complete -c nvim --condition 'string match -q "scp:*" -- (commandline -ct)' -a '(_do_scp_completion)'
```

If the current token starts with "scp:" make it "do" scp completions, in some form.

That function `_do_scp_completion` would have to:

- Take the current token
- Split it into protocol, user@host and file part (the syntax according to `:help netrw-path` is `vim scp://user@host/somefile`, which would be `string split -m 3 /`)
- Pass the user@host and the path to `ssh`, like the scp completions do.

**Comment by AndreaHasani** [comment]
1. I understand the point about chatgpt, first tried to ask it to add it to the scp.fish so I could undearstand the --condition syntaax but it failed on that. After that I mostly use it to get some guides which works quite well.

Anyhow, what would be a good workflow so I can debug the complete while I work on adding it. Since I don't know the fish syntax I have to add printf statements to see what the function is getting etc.

I'm trying to do this with a nvim_scp.fish file which I store on `~/.config/fish/completions/` but after sourcing that, I need to always reboot fish for complete to get the new function (and somehow printf also does not work sometimes)

**Comment by zanchey** [comment]
Probably your best option is to create `~/.config/fish/completions/nvm.fish` with `source /usr/share/fish/completions/nvm.fish` at the top and your new completions following it.
